torchrl/objectives/value/functional.py

```diff
@@ -158,7 +159,16 @@ def _fast_vec_gae(
     num_per_traj = _get_num_per_traj(done)
     td0_flat = _split_and_pad_sequence(td0, num_per_traj)
 
-    gammalmbdas = torch.ones_like(td0_flat[0])
+    device = done.device
+    if not isinstance(gammalmbda, torch.Tensor):
+        gammalmbda_tensor = torch.tensor(gammalmbda, device=device)
+    else:
+        gammalmbda_tensor = gammalmbda
+    lim = (
+        (torch.tensor(thr, device=device).log() / gammalmbda_tensor.log()).int().item()
```

Comment:
I guess the `.item()` can be dropped.

In case this code runs on a GPU and gammalmbda is a scalar: would this code not first have to bring single numbers to the GPU do the logarithms and division and bring it back to host memory immediately?
It may introduce a memory barrier, but I've not enough knowledge of the pytorch internals to be sure.

Reply:
yeah that's right
In practice we would not "bring" anything to gpu but directly create it on gpu. But then we'd bring it on cpu, that's for sure. 
probably negligeable but defo better without it